### ISSUE TITLE
No script tag injected in HTML file

### ISSUE THREAD

**Issue by ghost** [Densaugeo/reloadserver#1]
As the title says, the script tag that's supposed to be injected into HTML files is not there.

I noticed this error **once** (and never again after that, however the problem persisted, so I doubt it has anything to do with this but I'm adding it just in case):
```
----------------------------------------
Exception occurred during processing of request from ('127.0.0.1', 46302)
Traceback (most recent call last):
  File "/usr/lib/python3.11/socketserver.py", line 691, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/lib/python3.11/socketserver.py", line 361, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/lib/python3.11/http/server.py", line 667, in __init__
    super().__init__(*args, **kwargs)
  File "/usr/lib/python3.11/socketserver.py", line 755, in __init__
    self.handle()
  File "/usr/lib/python3.11/http/server.py", line 432, in handle
    self.handle_one_request()
  File "/usr/lib/python3.11/http/server.py", line 420, in handle_one_request
    method()
  File "/home/kwyntes/.local/pipx/venvs/reloadserver/lib/python3.11/site-packages/reloadserver/__init__.py", line 73, in do_GET
    super().do_GET()
  File "/usr/lib/python3.11/http/server.py", line 674, in do_GET
    self.copyfile(f, self.wfile)
  File "/home/kwyntes/.local/pipx/venvs/reloadserver/lib/python3.11/site-packages/reloadserver/__init__.py", line 39, in copyfile_interceptor
    outputfile.write(source.read().replace(b'</html>', SCRIPT_TAG + b'</html>'))
  File "/usr/lib/python3.11/socketserver.py", line 834, in write
    self._sock.sendall(b)
BrokenPipeError: [Errno 32] Broken pipe
----------------------------------------
```

Reloading does work properly in the directory listing though.

**Comment by Densaugeo** [comment]
Can you post the file you're seeing the issue with?

It decides whether to inject the script tag based on the content-type header (as determined by Python's http.server), and uses the closing </html> tag to determine where to insert the script. If either the content-type is incorrect or the closing tag is missing, the script won't be injected.

**Comment by ghost** [comment]
Ah I just found out the file was indeed missing a closing </html> tag, so that fixes it.

It would be nice however to have the script tag be appended/prepended to the file in case of a malformed document such as this one though, since then the reloading doesn't break whenever something messes up the HTML file (bad codegeneration for example).
I suppose this could be implemented simply by doing a regex replace like `/(<\/html>)?\s*$/` -> `<script>...</script>$1`.

**Comment by Densaugeo** [comment]
Not a bad idea. The other option would be to print an error message when that happens.

Which do you think would have been more useful in your case?

**Comment by ghost** [comment]
A warning that the file contains malformed HTML (or at least a missing closing </html> tag) while still injecting the script tag would be a nice in-between I think.

**Comment by Densaugeo** [comment]
Fixed by #4
